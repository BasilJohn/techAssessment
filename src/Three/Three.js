/**
 * @format
 * @flow strict-local
 */
import React, {Component} from 'react';
import {View, Text, TouchableOpacity} from 'react-native';
import {styles} from './styles';
import {useSelector} from 'react-redux';
import SlidingButton from '../customComponent/slidingButton';
import {NavigationScreenProps} from '@react-navigation/native';
import ButtonComponent from '../customComponent/buttonComponent';
import {higherOrderComponent} from '../hoc/buttonHOCComponent';


const VaritionOne=higherOrderComponent(ButtonComponent,styles.buttonTwoStyle);
const VaritionTwo=higherOrderComponent(ButtonComponent,styles.buttonThreeStyle,styles.whiteText);

type ThreeProps = NavigationScreenProps & {};
const Three = (props: ThreeProps) => {
  const value = useSelector(state => state.main.inputValue);
  const slideEndAction = () => {
    props.navigation.navigate('One');
  };

 
  return (
    <View style={styles.container}>
      <View style={styles.welcomeViewContainer}>
        <Text style={styles.inputValueStyle}>Input Data :{value}</Text>
      </View>
      <View style={styles.buttonContainer}>
        <TouchableOpacity
          onPress={() => props.navigation.navigate('One')}
          style={styles.button}>
          <Text style={styles.textStyle}>Go to screen one</Text>
        </TouchableOpacity>
      </View>
      <View style={styles.buttonContainer}>
        <TouchableOpacity
          onPress={() => props.navigation.navigate('Two')}
          style={styles.button}>
          <Text style={styles.textStyle}>Go to screen two</Text>
        </TouchableOpacity>
      </View>
      <View style={styles.variationButtonContainer}>
        <Text style={styles.variationHeaderText}>4 variations of a button</Text>
        {/* <View style={[styles.buttonStyle]}>
          <TouchableOpacity style={styles.button}>
            <Text style={styles.buttonTextStyle}>Press me</Text>
          </TouchableOpacity>
        </View> */}
        <ButtonComponent />
        {/* <View style={[styles.buttonStyle, styles.buttonTwoStyle]}>
          <TouchableOpacity style={styles.button}>
            <Text style={styles.buttonTextStyle}>Press me</Text>
          </TouchableOpacity>
        </View> */}
        <VaritionOne/>
        {/* <View style={[styles.buttonStyle, styles.buttonThreeStyle]}>
          <TouchableOpacity style={styles.button}>
            <Text style={styles.whiteText}>Press me</Text>
          </TouchableOpacity>
        </View> */}
        <VaritionTwo/>
        <View style={[styles.slidingButtonStyle]}>
          <SlidingButton testID={'slider'} endAction={slideEndAction}></SlidingButton>
        </View>
      </View>
    </View>
  );
};

export default Three;
