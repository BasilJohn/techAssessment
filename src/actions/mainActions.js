/**
 * @format
 * @flow strict-local
 */

export const setInputValue = (inputValue: string) => ({
  type: 'SET_INPUT_VALUE',
  payload: inputValue,
});

export const setEmulatorInformation = (isEmulator: string) => ({
  type: 'SET_EMULATOR_INFO',
  payload: isEmulator,
});
