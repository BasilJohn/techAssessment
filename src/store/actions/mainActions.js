/**
 * @format
 * @flow strict-local
 */

const setInputValue = (inputValue: string) => ({
  type: 'SET_INPUT_VALUE',
  payload: inputValue,
});

const setEmulatorInformation = (isEmulator: string) => ({
  type: 'SET_EMULATOR_INFO',
  payload: isEmulator,
});

export { setInputValue, setEmulatorInformation };
