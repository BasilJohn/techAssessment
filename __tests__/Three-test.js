/**
 * @format
 */

import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import {shallow} from 'enzyme';
import Three from '../src/Three/Three';
import mainReducer from '../src/reducers/reducer';
import configureMockStore from 'redux-mock-store';
import {setInputValue} from '../src/actions/mainActions';
import {NativeModules} from 'react-native';
import {Provider} from 'react-redux';
import {fireEvent, render, waitFor} from '@testing-library/react-native';
import * as redux from 'react-redux';

const mockStore = configureMockStore([]);
const store = mockStore({
  main: {
    inputValue: '',
    isEmulator: 'false',
  },
});

const comp = renderer.create(
  <Provider store={store}>
    <Three />
  </Provider>
);
test('Renders Component Three Correctly', () => {
  const tree = comp.toJSON();
  expect(tree).toMatchSnapshot();
});

describe('Check Value', () => {

    const props = {
        navigation: {
          navigate: jest.fn(),
        },
        setInputValue: jest.fn(),
      };

  it('Check Style', () => {
    const navigate = jest.fn();
    const wrapper = renderer.create(<Provider store={store}><Three {...props} /></Provider>);
    expect(wrapper.toJSON().children[3].children[4].children[0].children[0].children[1].props.style).toEqual({
      alignItems: 'center',
      justifyContent: 'center',
      borderRadius: 10,
      height: '100%',
      backgroundColor: 'rgba(110, 177, 247, 0.8)',
      width: '20%'
    });
 
  });
});
