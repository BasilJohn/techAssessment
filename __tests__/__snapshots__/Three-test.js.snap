// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Component Three Correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#1B1A24",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "flex-end",
        "height": 66.7,
        "justifyContent": "center",
        "paddingRight": 20,
        "width": "100%",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#FFFFFF",
        }
      }
    >
      Input Data :
      
    </Text>
  </View>
  <View
    style={
      Object {
        "alignSelf": "center",
        "backgroundColor": "grey",
        "height": 133.4,
        "marginBottom": 10,
        "width": "90%",
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "height": "100%",
          "justifyContent": "center",
          "opacity": 1,
          "width": "100%",
        }
      }
    >
      <Text
        style={
          Object {
            "textAlign": "center",
          }
        }
      >
        Go to screen one
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "alignSelf": "center",
        "backgroundColor": "grey",
        "height": 133.4,
        "marginBottom": 10,
        "width": "90%",
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "height": "100%",
          "justifyContent": "center",
          "opacity": 1,
          "width": "100%",
        }
      }
    >
      <Text
        style={
          Object {
            "textAlign": "center",
          }
        }
      >
        Go to screen two
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "width": "100%",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#B2BE18",
          "textAlign": "center",
        }
      }
    >
      4 variations of a button
    </Text>
    <View
      style={
        Array [
          Object {
            "alignSelf": "center",
            "borderRadius": 10,
            "height": 93.38000000000001,
            "marginBottom": 10,
            "width": "90%",
          },
          undefined,
        ]
      }
    >
      <View
        accessible={true}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "height": "100%",
            "justifyContent": "center",
            "opacity": 1,
            "width": "100%",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#6E9ED0",
                "textAlign": "center",
              },
              undefined,
            ]
          }
          testID="btnText"
        >
          Press me
        </Text>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "alignSelf": "center",
            "borderRadius": 10,
            "height": 93.38000000000001,
            "marginBottom": 10,
            "width": "90%",
          },
          Object {
            "backgroundColor": "#34434A",
          },
        ]
      }
    >
      <View
        accessible={true}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "height": "100%",
            "justifyContent": "center",
            "opacity": 1,
            "width": "100%",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#6E9ED0",
                "textAlign": "center",
              },
              undefined,
            ]
          }
          testID="btnText"
        >
          Press me
        </Text>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "alignSelf": "center",
            "borderRadius": 10,
            "height": 93.38000000000001,
            "marginBottom": 10,
            "width": "90%",
          },
          Object {
            "backgroundColor": "#6EB1F7",
          },
        ]
      }
    >
      <View
        accessible={true}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "height": "100%",
            "justifyContent": "center",
            "opacity": 1,
            "width": "100%",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#6E9ED0",
                "textAlign": "center",
              },
              Object {
                "color": "#FFFFFF",
                "textAlign": "center",
              },
            ]
          }
          testID="btnText"
        >
          Press me
        </Text>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "marginTop": 30,
            "width": "100%",
          },
        ]
      }
    >
      <View
        style={
          Object {
            "flex": 1,
            "justifyContent": "center",
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "alignSelf": "center",
                "backgroundColor": "#1B1A24",
                "borderColor": "#3E464E",
                "borderRadius": 10,
                "borderWidth": 1,
                "height": 60,
                "width": "90%",
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "height": "100%",
                  "justifyContent": "center",
                  "position": "absolute",
                  "width": "100%",
                },
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#FFFFFF",
                    "textAlign": "center",
                  },
                ]
              }
              testID="txtSlideText"
            >
              Slide me to continue
            </Text>
          </View>
          <View
            onMoveShouldSetResponder={[Function]}
            onMoveShouldSetResponderCapture={[Function]}
            onResponderEnd={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderReject={[Function]}
            onResponderRelease={[Function]}
            onResponderStart={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            onStartShouldSetResponderCapture={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "rgba(110, 177, 247, 0.8)",
                "borderRadius": 10,
                "height": "100%",
                "justifyContent": "center",
                "width": "20%",
              }
            }
          >
            <View
              style={
                Object {
                  "alignSelf": "flex-end",
                  "backgroundColor": "#6EB1F7",
                  "borderRadius": 10,
                  "height": "100%",
                  "justifyContent": "center",
                  "overflow": "hidden",
                  "width": 133.92857142857144,
                }
              }
            >
              <Text
                allowFontScaling={false}
                style={
                  Array [
                    Object {
                      "color": undefined,
                      "fontSize": 20,
                    },
                    Object {
                      "color": "#FFFFFF",
                      "textAlign": "center",
                    },
                    Object {
                      "fontFamily": "FontAwesome",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
              >
                ïˆ™
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
